app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt

```diff
@@ -636,24 +637,28 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
                 true
             }
             else -> {
-                // In case we're displaying search results, we wouldn't have navigated to home, and
-                // so we don't need to navigate "back to" browser fragment.
-                // See mirror of this logic in BrowserToolbarController#handleToolbarClick.
-                if (store.state.searchTerms.isBlank()) {
-                    val args by navArgs<SearchDialogFragmentArgs>()
-                    args.sessionId?.let {
-                        findNavController().navigate(
-                            SearchDialogFragmentDirections.actionGlobalBrowser(null),
-                        )
-                    }
-                }
-                view?.hideKeyboard()
-                dismissAllowingStateLoss()
+                goBack()
                 true
             }
         }
     }
 
+    private fun goBack() {
+        // In case we're displaying search results, we wouldn't have navigated to home, and
+        // so we don't need to navigate "back to" browser fragment.
+        // See mirror of this logic in BrowserToolbarController#handleToolbarClick.
+        if (store.state.searchTerms.isBlank()) {
+            val args by navArgs<SearchDialogFragmentArgs>()
+            args.sessionId?.let {
+                findNavController().navigate(
+                    SearchDialogFragmentDirections.actionGlobalBrowser(null),
+                )
+            }
+        }
```

Comment:
nit: If the above comment does not apply to the entire block I think it's useful to have an empty line to make that clear - so here would a new line before 658 - or if it does it could be a fun description instead